hack/externalnode/install-vm.ps1

```diff
@@ -167,41 +174,102 @@ function UpdateAgentConf() {
-    log "Creating service $AntreaAgent $AntreaAgentPath $AntreaAgentArgs"
+function CreateAntreaAgentStartupScript() {
+    $Script:StartAntreaAgentScript = "$AntreaAgentConfDir\Start-AntreaAgent.ps1"
+    $StartAntreaAgentScriptContent = '
+Param(
+    [parameter(Mandatory = $true)] [string] $OVSBridge,
+    [parameter(Mandatory = $true)] [string] $InstallDir
+)
+
+$AntreaSwitch = "antrea-switch"
+$AntreaAgentConfDir = [io.path]::combine($InstallDir, "conf")
+$AntreaAgentLogDir = [io.path]::combine($InstallDir, "logs")
+$AntreaAgentConfPath = [io.path]::combine($AntreaAgentConfDir, "antrea-agent.conf")
+$AntreaAgentLogFile = [io.path]::combine($AntreaAgentLogDir, "antrea-agent.log")
+$AntreaAgentPath = [io.path]::combine($InstallDir, "antrea-agent.exe")
+$InstallLog = [io.path]::combine($AntreaAgentLogDir, "install_vm.log")
+
+function Log($Info) {
+    $time = $(get-date -Format g)
+    "$time $Info " | Tee-Object $InstallLog -Append | Write-Host
+}
+
+function ClearOVSConfig() {
+    Log "Deleting OVS bridge $OVSBridge"
     try {
-        # Configured to auto-restart upon reboot
-        & nssm install $AntreaAgent $AntreaAgentPath $AntreaAgentArgs
-    } catch {
-        log "Failed to create service for $AntreaAgent, rc $_"
+        $adapterName = (Get-VMNetworkAdapter -ComputerName $(hostname.exe) -SwitchName $AntreaSwitch -ManagementOS).Name
+        ovs-vsctl.exe del-br $OVSBridge
+    }  catch {
+        Log "Failed to get VMSwitch $AntreaSwitch, rc $_"
         exit 1
     }
-}
 
-function StartAntreaAgentService()
-{
     try {
-        & nssm start $AntreaAgent
+        Remove-VMSwitch -ComputerName $(hostname.exe) $AntreaSwitch  -Force
     } catch {
-        log "Failed to start service for $AntreaAgent, rc $_"
+        Log "Ignore error while removing VMSwitch, rc $_"
+    }
+
+    try {
+        Rename-NetAdapter -Name "$adapterName~" -NewName "$adapterName"
+    }  catch {
+        Log "Failed to rename network adapter $adapterName~ to $adapterName, rc $_"
         exit 1
     }
 }
 
-CheckPrerequisites
+function CheckOVSConfigAndCleanup() {
+    $bridges = ovs-vsctl list-br
+    foreach ($br in $bridges) {
+        if ($br -ne $OVSBridge) {
+            continue
+        }
+        $ports = ovs-vsctl list-ports $OVSBridge
+        foreach ($port in $ports) {
+            $output = ovs-vsctl --no-headings --columns=error list interface "$port"
+            if ($output -ne "[]") {
+                ClearOVSConfig
+                break
+            }
+        }
+    }
+}
+
+function StartAntreaAgent() {
+    $antreaAgentArgs = "--config $AntreaAgentConfPath --log_file $AntreaAgentLogFile --logtostderr=false"
+    $cmd = "$AntreaAgentPath $antreaAgentArgs"
+    Invoke-Expression $cmd
+}
+
+CheckOVSConfigAndCleanup
+StartAntreaAgent
+'
+    Set-Content -Path $StartAntreaAgentScript -Value $StartAntreaAgentScriptContent
+}
+
+function StartAntreaAgentService() {
```

Comment:
Maybe rename it as CreateAntreaAgentService and move line 267 (nssm start $AntreaAgent) out of this function, and then call `nssm start $AntreaAgent` at the bottom after calling CreateAntreaAgentService ?

Reply:
Sure. I will keep it like this.
function CreateAntreaAgentService() {}
function StartAntreaAgentService() {} or function StartService($name) {}